src/expressions/andExpressionChecker.js

```diff
@@ -0,0 +1,54 @@
+import { getDiscplLogger } from '../utils/logging_util'
+
+export class AndExpressionChecker {
+  /**
+   * Create an AndExpressionChecker
+   * @param {ServiceProvider} serviceProvider
+   */
+  constructor (serviceProvider) {
+    this.serviceProvider = serviceProvider
+    this.logger = getDiscplLogger()
+    this.expression = 'AND'
+  }
+
+  /**
```

Comment:
Given how these getters are there in every expression checker maybe it makes sense to extract that to a generic expression checker? Having a getter like this also seems like a bit of a java-ism, we could also do direct property access instead.

Reply:
I went with methods instead of properties because there are some cyclic dependencies but base class does indeed make sense.